### PR TITLE
upgrade azure sdk version

### PR DESCRIPTION
**Pull request by andyzhangx**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
upgrade azure sdk version

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
```console
hack/pin-dependency.sh github.com/Azure/azure-sdk-for-go v54.1.0+incompatible
hack/pin-dependency.sh github.com/Azure/go-autorest/autorest v0.11.17
hack/pin-dependency.sh github.com/Azure/go-autorest/autorest/adal v0.9.10
hack/pin-dependency.sh github.com/Azure/go-autorest/autorest/to v0.2.0
```

#### Does this PR introduce a user-facing change?

```release-note
upgrade azure sdk version
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
upgrade azure sdk version
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -42,7 +42,7 @@ require (
 	github.com/evanphx/json-patch v4.9.0+incompatible
 	github.com/fsnotify/fsnotify v1.4.9
 	github.com/go-bindata/go-bindata v3.1.1+incompatible
-	github.com/go-openapi/spec v0.20.3
+	github.com/go-openapi/spec v0.19.5
```
Comment (dims) [inline go.mod:45]: this is going backwards @andyzhangx . 

Also i don't see anything related to the azure deps :)

**Comment by dims** [comment]
looks like a bunch of things to fix to get the CI jobs green @andyzhangx

**Comment by dims** [comment]
@andyzhangx please refresh this? (slack chatter on #k8s-code-organization if you need a hand)

File: go.mod [hunk 2]
```diff
@@ -10,7 +10,7 @@ go 1.16
 
 require (
 	bitbucket.org/bertimus9/systemstat v0.0.0-20180207000608-0eeff89b0690
-	github.com/Azure/azure-sdk-for-go v53.1.0+incompatible
+	github.com/Azure/azure-sdk-for-go v54.1.0+incompatible
```
Comment (liggitt) [inline go.mod:13]: /close

in favor of https://github.com/kubernetes/kubernetes/pull/102441 which updates to v55

**Pull request closed without merging**